cumulus_library/actions/builder.py

```diff
@@ -136,13 +158,31 @@ def run_protected_table_builder(
 
 
 def _run_workflow(
-    config: base_utils.StudyConfig, manifest: study_manifest.StudyManifest, filename: str
-) -> None:
+    config: base_utils.StudyConfig,
+    manifest: study_manifest.StudyManifest,
+    filename: str,
+    prepare: str,
+    data_path: pathlib.Path,
+    query_count: int,
+) -> int:
     """Loads workflow config from toml definitions and executes workflow
 
     :param config: a StudyConfig object
     :param manifest: a StudyManifest object
+    :param filename: Filename of the workflow config
+    :param prepare: If true, will render query instead of executing
+    :param data_path: If prepare is true, the path to write rendered data to
+    :param query_count: if prepare is true, the number of queries already rendered
+    :returns: a count of processed queries
     """
+    toml_path = pathlib.Path(f"{manifest._study_path}/{filename}")
+    if prepare:
+        with open(toml_path) as file:
```

Comment:
Interesting... I am use to the pylint check to always pass an `encoding=` arg to `open` (usually `encoding='utf8'`). Ruff isn't doing that here? I might be interested in seeing which package that's in and adding those checks, since that seems like a real issue (not high risk, but not zero) that can be hard to catch on dev machines and hard to debug later.

Reply:
yeah ruff didn't flag that, i'll see if i can find the right rule for it.

Reply:
https://docs.astral.sh/ruff/rules/unspecified-encoding/ - since its marked unstable, i'm inclined to not do it for now.